flybywiresim/aircraft
# feat(GPU): Implemented more realistic GPU handling

Fixes #[issue_no]

## Summary of Changes

This PR intends to make the GPU handling on both aircraft more realistic and immersive. 
It mainly does that by decoupling the GPU connection state from the MSFS vars.

As a result you will now be able to disconnect the GPU from the plane even when on a powered stand.

I also fixed an issue in the version checker for the a32nx.

In order to facilitate communication from the EFB via the eventbus to the extrashost, I integrated an eventbus instance into the EFB Wrapper

## Screenshots (if necessary)



## References





## Additional context



Discord username (if different from GitHub):\_chaoz_

## Testing instructions


Since this PR implements the same function for both planes the tests will be the same:

1. spawn C/D at a gate that you know has power 
2. after spawning check that the EXT Power buttons show AVAIL
3. check that the EFB Ground page shows GPU connected
4. click the GPU button on the EFB page
5. AVAIL lights should turn off
6. click it again
7. AVAIL lights should come back on
8. request pushback through some means
9. pushback
10. AVAIL lights should turn off the moment you start moving
11. Go back to the main menu and start a new flight at a stand that you know has no power
12. after spawning check that the EXT Power buttons dont show AVAIL
13. check that the EFB Ground page shows GPU disconnected
14. click the GPU button on the EFB page
15. button should turn orange and the msfs GPU cart should be called
16. if airport has a GPU cart then it should evenetually arrive
17. once it is hooked up the button should turn green and the EXT Power buttons should show AVAIL
18. click GPU button on EFB again
19. GPU cart should be sent away and AVAIL lights should turn off


## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo**, **flybywire-aircraft-a380-842 (4K)** or **flybywire-aircraft-a380-842 (8K)** download link at the bottom of the page
